cli/run_util.go

```diff
@@ -426,18 +430,45 @@ func (r WorkflowRunner) executeStep(
 	logger := log.NewLogger(opts)
 	stdout := logwriter.NewLogWriter(logger)
 
-	name := cmdArgs[0]
+	var name string
 	var args []string
-	if len(cmdArgs) > 1 {
-		args = cmdArgs[1:]
+	var envs []string
+
+	if workflow.Container.Image != "" {
```

Comment:
I think we should validate the `container` block a bit better. The image could be an empty string while other container fields are not (for example: `image: $MY_DOCKER_IMAGE` and the env is undefined).
Is the image field the bare minimum that needs to be filled? In this case, I think we could add a warning if image is missing, but other container fields are present.

Reply:
The image is the only parameter that is "required", if the customer leaves that empty everything else is meaningless so we will ignore that instead of failure. We didn't think it was necessary to handle this proactively as we do not expect anyone to start adding a container field without actually setting an image there.

Reply:
That's my point, someone adds both, but with an indirection that breaks at some point:
```yaml
container:
  image: $DOCKER_IMAGE
  ports: ["3000:3000"]
```
But I'm okay with the current state of the PR if image is the only required fields, just please collect these improvements somewhere (it's probably related to the `bitrise validate` extension)